pkg/util/etcdutil/migrate.go

```diff
@@ -136,41 +140,50 @@ func createMigratedEtcdCluster(restclient restclient.Interface, host, podIP stri
 	return nil
 }
 
-func checkEtcdClusterUp() error {
+func waitEtcdClusterRunning(restclient restclient.Interface, host string) error {
 	glog.Infof("initial delaying (30s)...")
 	time.Sleep(30 * time.Second)
 
-	// The checking does:
-	// - Trying to talk to etcd cluster via etcd service. The assumption here is that
-	//   the etcd service only selects the etcd pods newly created, excluding the boot one.
-	//   Once we can talk to it, we are sure that etcd cluster is created.
-	// - Then we list members and see if it's been reduced to 1 member. Because when we
-	//   can talk to the etcd cluster, we are certain there are 2 members at the beginning,
-	//   and will reduce to 1 eventually. That's the timeline of expected events.
-	//   As long as 1 member cluster is reached, we are certain cluster has been migrated successfully.
-	err := wait.PollImmediate(10*time.Second, 60*time.Second, func() (bool, error) {
-		cfg := clientv3.Config{
-			Endpoints:   []string{fmt.Sprintf("http://%s:2379", etcdServiceIP)},
-			DialTimeout: 5 * time.Second,
+	err := wait.Poll(10*time.Second, waitEtcdClusterRunningTime, func() (bool, error) {
+		res := restclient.Get().RequestURI(makeEtcdClusterURI(host, etcdClusterName)).Do()
+		if res.Error() != nil {
+			return false, res.Error()
 		}
-		etcdcli, err := clientv3.New(cfg)
-		if err != nil {
-			glog.Errorf("fail to create etcd client, retrying...: %v", err)
-			return false, nil
+		fmt.Println("haha")
```

Comment:
I don't think you can right now. @pbx0 is looking into general failures right now -- it may be unrelated to this PR